packages/amplication-git-utils/src/git/github.service.ts

```diff
@@ -306,31 +322,471 @@ export class GithubService implements GitProvider {
+    const { sha } = await this.getFirstDefaultBranchCommit(
+      owner,
+      repositoryName
+    );
+    const refs = await this.getBranch(owner, repositoryName, branchName);
+
+    if (refs.sha !== sha) {
+      return this.createBranch(owner, repositoryName, branchName, sha);
+    }
+
+    return refs;
+  }
+
+  async createCommit({
+    owner,
+    repositoryName,
+    commitMessage,
+    branchName,
+    changes,
+  }: CreateCommitArgs): Promise<void> {
+    const lastCommit = await this.getLastCommit(
+      owner,
+      repositoryName,
+      branchName
+    );
+
+    console.log(`Got last commit with url ${lastCommit.html_url}`);
+
+    if (!lastCommit) {
+      throw new Error("No last commit found");
+    }
+
+    const lastTreeSha = await this.createTree(
+      owner,
+      repositoryName,
+      lastCommit.sha,
+      lastCommit.commit.tree.sha,
+      changes
+    );
+
+    console.info(`Created tree for for ${owner}/${repositoryName}`);
+
+    const { data: commit } = await this.octokit.rest.git.createCommit({
+      message: commitMessage,
+      owner,
+      repo: repositoryName,
+      tree: lastTreeSha,
+      parents: [lastCommit.sha],
+    });
+
+    console.info(`Created commit for ${owner}/${repositoryName}`);
+
+    await this.octokit.rest.git.updateRef({
+      owner,
+      repo: repositoryName,
+      sha: commit.sha,
+      ref: `heads/${branchName}`,
+    });
+
+    console.info(`Updated branch ${branchName} for ${owner}/${repositoryName}`);
+  }
+
+  async getPullRequestForBranch({
+    owner,
+    repositoryName,
+    branchName,
+  }: GetPullRequestFomBranchArgs): Promise<
+    { url: string; number: number } | undefined
+  > {
+    const branchInfo: {
+      repository: {
+        ref: {
+          associatedPullRequests: {
+            edges: [
+              {
+                node: {
+                  url: string;
+                  number: number;
+                };
+              }
+            ];
+          };
+        };
+      };
+    } = await this.octokit.graphql(
+      `
+          query ($owner: String!, $repo: String!, $head: String!) {
+            repository(name: $repo, owner: $owner) {
+              ref(qualifiedName: $head) {
+                associatedPullRequests(first: 1, states: OPEN) {
+                  edges {
+                    node {
+                      id
+                      number
+                      url
+                    }
+                  }
+                }
+              }
+            }
+          }`,
+      {
+        owner,
+        repo: repositoryName,
+        head: branchName,
+      }
+    );
+
+    const existingPullRequest =
+      branchInfo.repository.ref?.associatedPullRequests?.edges?.[0]?.node;
+
+    return existingPullRequest;
+  }
+
+  async createPullRequestForBranch({
+    owner,
+    repositoryName,
+    branchName,
+    pullRequestTitle,
+    pullRequestBody,
+    defaultBranchName,
+    pullRequestUrl,
```

Comment:
We dont need this param, if there is a pull request already we don't need to get here any way

Reply:
l don't understand so lets talk about it